src/components/TokensCard/TokensCard.tsx

```diff
@@ -23,7 +23,7 @@ export type TTokensCard = {
 };
 
 export const TokensCard: FC<TTokensCard> = ({ collectionId, tokenId, testid, ...props }) => {
-  const [token, setToken] = useState<(NFTToken & Partial<Offer>) | undefined>(props.token);
+  const [token, setToken] = useState<(NFTToken & NFTCollection & Partial<Offer>) | undefined>(props.token);
```

Comment:
lets refactor it a bit:
1. Token type should strictly be "TokenByIdResult" from SDK
2. can't be undefined
3. Whenever this component is called - prepare correct type in parent (SDK and back-end offers are always hold token in that type)

Also, pay attention to similar cases in other places.